packages/flutter/lib/src/services/system_chrome.dart

```diff
@@ -333,6 +333,63 @@ class SystemChrome {
   ///   return /* ... */;
   /// }
   /// ```
+  ///
+  /// For more complex control of the system overlay styles, consider using
+  /// an [AnnotatedRegion] widget instead of calling [setSystemUiOverlayStyle]
+  /// directly. This widget places a value directly into the layer tree where
+  /// it can be hit-tested by the framework. On every frame, the framework will
+  /// select the first annotated region it finds under the status and
+  /// navigation bar and synthesize them into a single style. This process can
+  /// be short-circuted by passing `sized: false` to a single annotated region
+  /// widget.
+  ///
+  /// ## Sample Code
```

Comment:
Please use `{@tool snippet --template=stateful_widget }...{@end-tool}` instead of "## Sample code" for this kind of sample.  You'll leave out everything except for the contents of the State object, like so:

```dart
  ///  {@tool snippet --template=stateful_widget }
  /// The following example creates a widget that changes the status bar color
  /// to a random value on Android.
  ///
  /// ```dart
  ///    final _random = math.Random();
  ///    SystemUiOverlayStyle _currentStyle = SystemUiOverlayStyle.light;
  ///
  ///    void _changeColor() {
  ///      final color = Color.fromRGBO(
  ///        _random.nextInt(255),
  ///        _random.nextInt(255),
  ///        _random.nextInt(255),
  ///        1.0,
  ///      );
  ///      setState(() {
  ///        _currentStyle = SystemUiOverlayStyle.dark.copyWith(
  ///          statusBarColor: color,
  ///        );
  ///      });
  ///    }
  ///
  ///    @override
  ///    Widget build(BuildContext context) {
  ///      return Scaffold(
  ///        body: AnnotatedRegion(
  ///          sized: false,
  ///          value: _currentStyle,
  ///           child: Center(
  ///             child: RaisedButton(
  ///               child: const Text('Change Color'),
  ///               onPressed: _changeColor,
  ///             ),
  ///           ),
  ///         ),
  ///       );
  ///     }
  /// {@end-tool}
```

Reply:
How do I add imports? This doesn't know about dart:math or SystemUiOverlayStyle

Reply:
Oh.  You review and approve this PR: https://github.com/flutter/flutter/pull/24830 :-)

I ran into the same problem in that PR and fixed it.  I'm happy to split that out into a separate PR since you need it.

Reply:
It's not critical to land this now, so I'm happy to wait. I will take a look at the PR as well tho!

Reply:
Here's the split out PR: https://github.com/flutter/flutter/pull/25184

Reply:
How do I place imports into the "code-imports" section?

Reply:
Add a section that looks like this:
```
    ```code-imports
    import 'package:foo/foo.html';
    ```
```